expression/builtin.go

```diff
@@ -173,9 +177,145 @@ func (b *baseBuiltinFunc) getArgs() []Expression {
 }
 
 func (b *baseBuiltinFunc) vecEval(input *chunk.Chunk, result *chunk.Column) error {
+	if b.self != nil && !b.vec {
+		return b.row2vec(input, result)
+	}
 	return errors.Errorf("baseBuiltinFunc.vecEval() should never be called, please contact the TiDB team for help")
 }
 
+func (b *baseBuiltinFunc) row2vec(input *chunk.Chunk, result *chunk.Column) error {
+	result.Reset()
+	it := chunk.NewIterator4Chunk(input)
+	sel := input.Sel()
+	if sel == nil {
+		switch b.tp.EvalType() {
+		case types.ETInt:
+			result.PreAllocInt64(input.NumEffectiveRows())
```

Comment:
s/ NumEffectiveRows/ NumRows

Reply:
Is always using `NumEffectiveRows` in vectorized evaluation more clear and safe?